baseapp/abci.go

```diff
@@ -792,6 +793,20 @@ func (app *BaseApp) flushCommit() storetypes.CommitID {
 	return commitID
 }
 
+// workingHash gets the apphash that will be finalized in commit.
+// These writes will be persisted to the root multi-store (app.cms) and flushed to
+// disk in the Commit phase. This means when the ABCI client requests Commit(), the application
+// state transitions will be flushed to disk and as a result, but we already have
+// an application Merkle root.
+func (app *BaseApp) workingHash() []byte {
+	app.finalizeBlockState.ms.Write()
```

Comment:
What's the reason for this call?

Reply:
since we moved the writing from comit to working hash now after Initchain & commit are called there is still no state, so we need to write to disk in order to call commit

Reply:
I think it's still a misnomer to call this `workingHash` as the name implies it's idempotent. But in reality, this now has side affects. Perhaps consider moving the flush/write to disk back to `Commit`.